public/app/features/manage-dashboards/state/actions.ts

```diff
@@ -281,7 +282,7 @@ export async function moveFolders(folderUIDs: string[], toFolder: FolderInfo) {
 }
 
 export function createFolder(payload: any) {
-  return getBackendSrv().post('/api/folders', payload);
+  return getFolderAPI().createFolder(payload);
 }
```

Comment:
The only place this function is used is in Alerting, and it actually should be deleted.

The rest of the frontend all makes their API requests through an RTK Query slice to properly handle caching and invalidation https://github.com/grafana/grafana/blob/joshhunt/deprecate-create-folder-fn/public/app/features/browse-dashboards/api/browseDashboardsAPI.ts#L112-L135

We'll need to come up with a way to swap out the API... which is much more difficult when the APIs accept and return differently shaped data...

Reply:
> The only place this function is used is in Alerting, and it actually should be deleted.

Good to know. I am able to make calls to the new endpoint and create a folder that way without any issue though so I am not sure how that's happening.

Reply:
From where in the UI? 🤔

Reply:
If I go to `/dashboards` and then click on the "New" button > "create folder"

Reply:
Hmmm interesting. Are you sure that's using the new K8s API? I just tested it by changing `createFolder` to throw an error:

```
export function createFolder(payload: any) {
  throw new Error("using deprecated function 'createFolder'");
  return getBackendSrv().post('/api/folders', payload);
}
```

and the folder was created, indicating this method is not called there. Looking at the code, I'm pretty sure it's using a different code path  (calling the RTK Query methods instead) https://github.com/grafana/grafana/blob/63841fefd64bb7bdd1d5d80a802c185ce6fa6153/public/app/features/browse-dashboards/components/CreateNewButton.tsx/#L27

Reply:
I'll double check, but I think so. Depending on whether I enable `kubernetesFolders` or not, the payload changes. For example, we haven't populated the "url" field yet and it doesn't appear when the feature is enabled.

How do I include a new API endpoint in the RTK Query methods? I searched for `api/folders` and only found references to the other endpoints.

Reply:
You're right. That's what it was. Now I need to reevaluate the changes this PR was trying to make. Moving this back to draft for now. Thank you very much for the input!